airflow/executors/executor_loader.py

```diff
@@ -344,8 +345,24 @@ def validate_database_executor_compatibility(cls, executor: type[BaseExecutor])
         if engine and engine.dialect.name == "sqlite":
             raise AirflowConfigException(f"error: cannot use SQLite with the {executor.__name__}")
 
+    @classmethod
+    def _warn_of_deprecated_executor(cls, executor_name: str) -> None:
+        """
+        Warn of deprecated executor.
+
+        :param executor_name: Name of the executor
+        """
+        warnings.warn(
+            f"\nThe use and support of the {executor_name} is deprecated and will be removed in Airflow 3.0.\n"
+            "Please migrate to using Multiple Executor Configuration instead:\n"
+            "https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/executor/index.html#using-multiple-executors-concurrently",
```

Comment:
If you want, you can chop the index.html out so it's a bit shorter:
```suggestion
            "https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/executor/#using-multiple-executors-concurrently",
```

Reply:
Nice, committed!